### DISCUSSION TITLE
what version of the clang complier is required for this project ?

### DISCUSSION THREAD

**Discussion by trriger** [doe300/VC4C#153] [General]
I install a clang complier which it  version is 11 ,but it can't make(cmake) this project.  It  seems that need a lower version.
And I try to change clang-9 but it occurs an another error. when itgo wrong , it's at 133 lines of cmakeList, but I dont know why it went wrong.

**Comment by doe300** [comment]
Currently clang versions 3.9 to 10 are supported. I will add support for clang/LLVM 11 as soon as I have a setup to test it. If you want to try it out, you can (in `cmake/clang.cmake`) for all list of program versions just add the version11 and see if it compiles correctly (e.g. add `clang-11` to line 6, `opt-11` to line 9 and so on). There is a slight chance that the compilation will then fail, if the LLVM API changed in a non-compatible way.

For any already supported clang/LLVM version you will need to install the LLVM base library, the clang compiler and the LLVM development packages (e.g. on Debian to install clang/LLVM 9, that would be `llvm-9`, `clang-9` and `llvm-9-dev`.
